python/src/intracktive/convert_OLD.py

```diff
@@ -0,0 +1,300 @@
+non_zero = tracks_to_tracks.nonzero()
+
+for i in range(len(non_zero[0])):
+    track_index = non_zero[1][i]
+    parent_track_index = direct_parent_index_map[track_index]
+    tracks_to_tracks[non_zero[0][i], non_zero[1][i]] = parent_track_index + 1
+
+# Convert to CSR format for efficient row slicing
+tracks_to_points = points_to_tracks.T.tocsr()
+points_to_tracks = points_to_tracks.tocsr()
+tracks_to_tracks = tracks_to_tracks.tocsr()
+
+print(f"Converted to CSR in {time.monotonic() - start} seconds")
+start = time.monotonic()
+
+# save the points array
+top_level_group = zarr.hierarchy.group(
+    zarr.storage.DirectoryStore(zarr_path.as_posix()),
+    overwrite=True,
+)
+
+points = top_level_group.create_dataset(
+    "points",
+    data=points_array,
+    chunks=(1, points_array.shape[1]),
+    dtype=np.float32,
+)
+points.attrs["values_per_point"] = num_values_per_point
+points.attrs["mean_x"] = mean_x
+points.attrs["mean_y"] = mean_y
+points.attrs["mean_z"] = mean_z
+points.attrs["extent_xyz"] = extent_xyz
+points.attrs["fields"] = ['z', 'y', 'x', 'radius'][:num_values_per_point]
+
+
+top_level_group.create_groups("points_to_tracks", "tracks_to_points", "tracks_to_tracks")
+# TODO: tracks_to_points may want to store xyz for the points, not just the indices
+# this would make the indices array 3x (4x?) larger, but would eliminate the need to
+# fetch coordinates again based on point IDs
+tracks_to_points_zarr = top_level_group["tracks_to_points"]
+tracks_to_points_zarr.attrs["sparse_format"] = "csr"
+tracks_to_points_zarr.create_dataset("indices", data=tracks_to_points.indices)
+tracks_to_points_zarr.create_dataset("indptr", data=tracks_to_points.indptr)
+tracks_to_points_xyz = np.zeros((len(tracks_to_points.indices), 3), dtype=np.float32)
+for i, ind in enumerate(tracks_to_points.indices):
+    t, n = divmod(ind, max_points_in_timepoint)
+    tracks_to_points_xyz[i] = points_array[t, num_values_per_point * n:num_values_per_point * (n + 1)][:3]
+# TODO: figure out better chunking?
+tracks_to_points_zarr.create_dataset(
+    "data",
+    data=tracks_to_points_xyz,
+    chunks=(2048, 3),
+    dtype=np.float32,
+)
+
+points_to_tracks_zarr = top_level_group["points_to_tracks"]
+points_to_tracks_zarr.attrs["sparse_format"] = "csr"
+points_to_tracks_zarr.create_dataset("indices", data=points_to_tracks.indices)
+points_to_tracks_zarr.create_dataset("indptr", data=points_to_tracks.indptr)
+
+tracks_to_tracks_zarr = top_level_group["tracks_to_tracks"]
+tracks_to_tracks_zarr.attrs["sparse_format"] = "csr"
+tracks_to_tracks_zarr.create_dataset("indices", data=tracks_to_tracks.indices)
+tracks_to_tracks_zarr.create_dataset("indptr", data=tracks_to_tracks.indptr)
+tracks_to_tracks_zarr.create_dataset("data", data=tracks_to_tracks.data)
+
+print(f"Saved to Zarr in {time.monotonic() - start} seconds")
+
+print(f"Full conversion took {time.monotonic() - start_fullBegin} seconds")
+
+
+# # This is what an example resulting Zarr store looks like:
```

Comment:
This information is pretty useful, so I just wanted to check it's available somewhere other than in `convert_OLD.py`, which some may ignore.

Reply:
Thanks for review @andy-sweet! One question, how do/did you print the tree structure of the zarr, including directory sizes? I tried playing around with `du` and `tree`, but cannot reproduce the output as presented here. (`du -sh tracks_bundle.zarr` just gives me: `520M	tracks_bundle.zarr`)

Reply:
@aganders3 wrote this, but my guess is they may have just written this manually or by editing the output of something like

```
tree -h -L 2 ZSNS001_tracks_bundle.zarr
```

Reply:
Yes, it was some combination of `tree` and `du` but manually combined.